qa/os/src/test/java/org/elasticsearch/packaging/test/PackagingTestCase.java

```diff
@@ -207,4 +185,87 @@ protected static Shell newShell() throws Exception {
         }
         return sh;
     }
+
+    /**
+     * Run the command to start Elasticsearch, but don't wait or test for success.
+     * This method is useful for testing failure conditions in startup. To await success,
+     * use {@link #startElasticsearch()}.
+     * @return Shell results of the startup command.
+     * @throws Exception when command fails immediately.
+     */
+    public Shell.Result runElasticsearchStartCommand() throws Exception {
+        switch (distribution.packaging) {
+            case TAR:
+            case ZIP:
+                return Archives.runElasticsearchStartCommand(installation, sh);
+            case DEB:
+            case RPM:
+                return Packages.runElasticsearchStartCommand(sh);
+            case DOCKER:
+                // nothing, "installing" docker image is running it
+        }
+        return Shell.NO_OP;
+    }
+
+    public void stopElasticsearch() throws Exception {
+        distribution().packagingConditional()
+            .forPackage(() -> Packages.stopElasticsearch(sh))
```

Comment:
We should probably keep the style consistent, either using packaging conditional, or a switch statement as in the previous method. The former has the advantage that adding any new packaging type will force us to add new cases, although we obviously aren't likely to add any, at least not soon or very often.

Reply:
I think that for the moment the switch statements are simpler — some of the switch statements return values. I could modify the packaging conditional to return a value, but I'd probably need to use generics for type safety. In other words, the switch statement is more convenient for side effects. And we can always re-introduce the conditional pattern if we need the enforcement of all cases. So I'll revert to switch statements here.